test/jdk/java/util/zip/ChecksumBase.java

```diff
@@ -69,6 +71,13 @@ private static void testWrappedByteBuffer(Checksum checksum, long expected) {
         checkChecksum(checksum, expected);
     }
 
+    private static void testWrappedMemorySegment(Checksum checksum, long expected) {
+        checksum.reset();
+        ByteBuffer bb = MemorySegment.ofArray(BYTES_123456789).asByteBuffer();
+        checksum.update(bb);
+        checkChecksum(checksum, expected);
+    }
```

Comment:
The src changes look right. For the tests then I wonder if we could expand to exercise with view of byte buffers on memory segments allocated from all arenas with bounded lifetime. I think that would help the test coverage.